sips/sip-x/sip-x-authentication-protocol.md

```diff
@@ -0,0 +1,397 @@
+
+## Authentication Response
+
+The authentication response is a signed JWT that contains the requested and authorized data. The token is signed with the public key of the request signature, i.e. the app transit key.
+
+The payload must contain the following claims:
+
+| Claim name        | Type    | Description                                                                                                                                                                                                         |
+| ----------------- | ------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| jti               | string  | As defined in RFC7519                                                                                                                                                                                               |
+| iat               | string  | As defined in RFC7519                                                                                                                                                                                               |
+| exp               | string  | As defined in RFC7519                                                                                                                                                                                               |
+| iss               | string  | Decentralized identifier defined in DID specification representing the user's account.                                                                                                                              |
+| private_key       | string  | Encrypted app private key for the provided domain. The key is encrypted with the public key of the app transit key.                                                                                                 |
+| public_keys       | array   | Single item array containing the public key of the selected account in hex representation.                                                                                                                          |
+| profile           | object  | Object containing properties of the users, the object schema should be well-known type of Appendix B. This can be the public profile of the selected account.                                                       |
+| apps_meta         | object  | Information about user data of different apps. Property names are the domain name of the app. Each property value is an object of containing properties for `storage` and `publicKey`. See "Application Meta Data". |
+| username          | string  | BNS username, owned by the first public key of `public_keys` claim. Can be the empty string                                                                                                                         |
```

Comment:
We're currently in the process of removing the `username` from the authentication flow completely. https://github.com/hirosystems/stacks.js/pull/1230 Keeping the username adds a bunch of complications and should be verified by the application anyway, so it makes sense for the application to fetch the username in the first place.

I bumped the token version to 1.4.0 for this, in case anybody is validating very strictly.

_some discussions/comments here:_
- https://github.com/hirosystems/stacks.js/pull/1230
- https://github.com/hirosystems/stacks-wallet-web/pull/2333

Reply:
+1 this seems like a good separation of concerns